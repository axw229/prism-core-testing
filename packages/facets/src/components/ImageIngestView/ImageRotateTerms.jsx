/* eslint-disable jsx-a11y/label-has-for */
// @flow
import React, { useState, useRef, useEffect, useContext } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { CircleLoader } from '@prism/toolkit'
import 'src/providers/fontawesome/fontawesome'
import { FormattedMessage, useIntl } from 'react-intl'
import 'src/scss/convenience/visually-hidden.scss'
import { KEY_CODES } from 'src/constants/globals'
import ConfigurationContext, { type ConfigurationContextType } from 'src/contexts/ConfigurationContext/ConfigurationContext'

import './ImageRotateTerms.scss'

const baseClass = 'image-rotate-terms-modal'
const wrapperClass = `${baseClass}__wrapper`
const wrapperContainerClass = `${wrapperClass}__container`
const wrapperContainerInactiveClass = `${wrapperContainerClass}--inactive`
const wrapperToolsClass = `${wrapperClass}__tools`
const wrapperToolsMessageClass = `${wrapperToolsClass}__message`
const wrapperToolsRotateArrowClass = `${wrapperToolsClass}__rotate-arrow`
const wrapperAgreeTermsClass = `${wrapperClass}__agree-terms`
const wrapperAgreeTermsCheckboxLabelClass = `${wrapperAgreeTermsClass}__checkbox-label`
const wrapperAgreeTermsTextClass = `${wrapperAgreeTermsClass}__text`
const wrapperAgreeTermsAcceptClass = `${wrapperAgreeTermsClass}__accept`
const wrapperAgreeTermsAcceptActiveClass = `${wrapperAgreeTermsAcceptClass}--active`
const wrapperLoaderClass = `${wrapperClass}__loader`

type Props = {
  rotateImage: Function,
  imageData: Object,
  handleDismiss: Function
}

const ImageRotateTerms = ({ rotateImage, imageData, handleDismiss }: Props) => {
  const [acceptTerms, setAcceptTerms] = useState(false)
  // eslint-disable-next-line no-unused-vars
  const [hideModal, setHideModal] = useState(false)
  let mouseDown = false
  const btnRef = useRef()
  const intl = useIntl()
  const { cvw: { termsOfUseLink = 'https://www.sherwin-williams.com/terms-of-use' } }: ConfigurationContextType = useContext(ConfigurationContext)

  useEffect(() => {
    btnRef.current.focus()
  }, [])

  function handleChange () {
    setAcceptTerms(!acceptTerms)
  }

  function mouseDownHandler () {
    mouseDown = true
  }

  function mouseUpHandler () {
    setTimeout(() => {
      mouseDown = false
    }, 200)
  }

  function focusHandler (e: Object) {
    if (mouseDown) {
      e.target.blur()
    }
  }

  function keyDownHandler (e: KeyboardEvent) {
    if (e.keyCode && (e.keyCode === KEY_CODES.KEY_CODE_ENTER || e.keyCode === KEY_CODES.KEY_CODE_SPACE)) {
      e.preventDefault()
      handleChange()
    }
  }

  return (
    <div className={`${wrapperClass}`}>
      <div className={`${wrapperContainerClass} ${hideModal ? `${wrapperContainerInactiveClass}` : ``}`}>
        <div className={`${wrapperToolsClass}`}>
          <div className={`${wrapperToolsMessageClass}`}>
            <FormattedMessage id='PREVIEW_ROTATE_SCALE' />
          </div>
          <button
            aria-label='rotate image 90 degree anticlockwise'
            ref={btnRef}
            onMouseDown={mouseDownHandler}
            onMouseUp={mouseUpHandler}
            onFocus={focusHandler}
            className={`${wrapperToolsRotateArrowClass}`} onClick={() => rotateImage(false)}><FontAwesomeIcon icon={['fal', 'undo']} size='xs' /></button>
          <button
            aria-label='rotate image 90 degree clockwise'
            onMouseDown={mouseDownHandler}
            onMouseUp={mouseUpHandler}
            onFocus={focusHandler}
            className={`${wrapperToolsRotateArrowClass}`} onClick={() => rotateImage(true)}><FontAwesomeIcon icon={['fal', 'redo']} size='xs' /></button>
        </div>
        <div className={`${wrapperAgreeTermsClass}`}>
          <div>
            <span>
              <label
                aria-label={intl.formatMessage({ id: 'ACCEPT_TERM' })}
                onMouseDown={mouseDownHandler}
                onMouseUp={mouseUpHandler}
                onFocus={focusHandler}
                onKeyDown={keyDownHandler}
                tabIndex='0' className={`${wrapperAgreeTermsCheckboxLabelClass}`}>
                {
                  (acceptTerms) ? <FontAwesomeIcon icon={['fa', 'dot-circle']} style={{ color: '#2cabe1' }} size='lg' />
                    : <FontAwesomeIcon icon={['fa', 'dot-circle']} style={{ color: '#e5e5e5' }} size='lg' />
                }
                <input tabIndex='-1' className='visually-hidden' type='checkbox' value='terms' checked={acceptTerms} onChange={handleChange} />
              </label>
            </span>
            <span className={`${wrapperAgreeTermsTextClass}`}>
              <FormattedMessage id='I_ACCEPT' /> <a href={termsOfUseLink} target='_blank'><FormattedMessage id='TERMS_OF_USE' /></a>
            </span>
          </div>
        </div>
        <button
          disabled={!acceptTerms}
          aria-label={acceptTerms ? 'done' : ''}
          tabIndex={acceptTerms ? '0' : '-1'}
          className={`${wrapperAgreeTermsAcceptClass} ${acceptTerms ? `${wrapperAgreeTermsAcceptActiveClass}` : ``}`}
          onClick={handleDismiss}><FormattedMessage id='DONE' /></button>
      </div>
      { hideModal && <div className={`${wrapperLoaderClass}`}><CircleLoader /></div> }
    </div>
  )
}

export {
  wrapperToolsMessageClass,
  wrapperToolsRotateArrowClass,
  wrapperAgreeTermsAcceptClass,
  wrapperAgreeTermsCheckboxLabelClass,
  wrapperAgreeTermsTextClass,
  wrapperAgreeTermsAcceptActiveClass
}
export default ImageRotateTerms
