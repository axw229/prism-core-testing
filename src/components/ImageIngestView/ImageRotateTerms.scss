.image-rotate-terms-modal {
  &__wrapper {
    align-content: center;
    bottom: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    left: 50%;
    margin-left: -9.5rem;
    position: absolute;
    right: 0;
    text-align: center;
    top: 0;
    width: 19rem;
    z-index: 1000;
    &__container {
      background: var(getThemeColor('primaryBg'));
      display: flex;
      flex-direction: column;
      line-height: 1.2em;
      &--inactive {
        display: none;
      }
    }
    &__loader {
      color: var(getThemeColor('secondary'));
      display: block;
      line-height: 1.2em;
      margin-top: 6px;
    }
    &__tools {
      margin-top: 6px;
      &__message {
        color: var(getThemeColor('black'));
        margin-bottom: 6px;
        padding: .75em;
      }
      &__rotate-arrow {
        background: var(getThemeColor('primaryBg'));
        border: 1px solid transparent;
        display: inline-block;
        font-size: 3em;
        height: 48px;
        margin-right: 10px;
        outline: none;
        padding: 0;
        width: 48px;
        &:focus {
          outline-color: var(getThemeColor('secondary'));
          outline-style: solid;
          outline-width: 2px;
        }
      }
    }
    &__agree-terms {
      height: 4em;
      padding: 1em;
      text-align: center;
      vertical-align: middle;
      &__checkbox-label {
        cursor: pointer;
        outline: none;
        &:focus {
          outline-color: var(getThemeColor('secondary'));
          outline-style: solid;
          outline-width: 2px;
        }
      }
      &__text {
        border: none;
        flex: 1 0 auto;
        width: 62%;
        padding: 0 0.5em;
        a {
          text-decoration: underline;
          color: var(getThemeColor('primary'))
        }
      }
      &__accept {
        background: var(getThemeColor('white'));
        border: 1px solid var(getThemeColor('light'));
        color: var(getThemeColor('light'));
        cursor: default;
        display: inline-block;
        flex: 1 0 auto;
        line-height: 1.25em;
        margin: -1em auto 1.75em;
        outline: none;
        padding: .75em 1.75em;
        text-align: center;
        width: auto;
        &--active {
          border: 1px solid var(getThemeColor('secondary'));
          color: var(getThemeColor('secondary'));
          cursor: pointer;
          &:focus {
            outline-style: solid;
            outline-color: var(getThemeColor('secondary'));
            outline-width: 2px;
          }
        }
      }
    }
  }
}
