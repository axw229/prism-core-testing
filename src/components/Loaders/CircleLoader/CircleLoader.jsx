// @flow
import React from 'react'

import './CircleLoader.scss'

type Props = {
  color?: string,
  className?: string
}

function CircleLoader (props: Props) {
  const { color, className, ...other } = props

  let circleProps = {}

  if (color) {
    circleProps = Object.assign({}, { stroke: color })
  }

  return (
    <svg className={`prism-loader-circle ${typeof className === 'string' ? className : ''}`} {...other} xmlns='http://www.w3.org/2000/svg' viewBox='0 0 100 100' preserveAspectRatio='xMidYMid'>
      <circle cx='50' cy='50' {...circleProps} fill='none' strokeWidth='5' r='35' strokeDasharray='164.93361431346415 56.97787143782138' transform='rotate(143.836 50 50)'>
        <animateTransform attributeName='transform' type='rotate' calcMode='linear' values='0 50 50;360 50 50' keyTimes='0;1' dur='1s' begin='0s' repeatCount='indefinite' />
      </circle>
    </svg>
  )
}

export default React.memo<Props>(CircleLoader)
