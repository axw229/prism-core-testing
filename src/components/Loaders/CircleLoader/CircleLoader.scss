@import 'src/scss/convenience/_keyframe-animations.scss';

.prism-loader-circle {
  $dashDuration: 1.8s;
  $times: 5;
  $rotateDuration: $dashDuration / 2;
  $strokeColorDuration: ($dashDuration * 2) * $times;

  @keyframes strokeColor {
    0% {
      stroke: var(getVarName('loaders.circle.color1'));
    }
    20% {
      stroke: var(getVarName('loaders.circle.color2'));
    }
    40% {
      stroke: var(getVarName('loaders.circle.color3'));
    }
    60% {
      stroke: var(getVarName('loaders.circle.color4'));
    }
    80% {
      stroke: var(getVarName('loaders.circle.color5'));
    }
    100% {
      stroke: var(getVarName('loaders.circle.color1'));
    }
  }

  @keyframes dashAdjust {
    0% {
      stroke-dasharray: var(getVarName('loaders.circle.beginDash')) var(getVarName('loaders.circle.beginGap'));
    }
    100% {
      stroke-dasharray: var(getVarName('loaders.circle.endDash')) var(getVarName('loaders.circle.endGap'));
    }
  }

  animation: fadeOpacityIn 1s, rotate $rotateDuration infinite linear;
  display: block;
  height: auto;
  margin: auto;
  max-height: 90%;
  max-width: 90%;
  opacity: 1;
  stroke: currentColor;
  width: 3em;

  &__circle {
    animation: strokeColor $strokeColorDuration infinite cubic-bezier(0.075, 0.820, 0.165, 1.000), dashAdjust $dashDuration infinite alternate cubic-bezier(0.390, 0.575, 0.565, 1.000);
    stroke-linecap: round;
  }
}