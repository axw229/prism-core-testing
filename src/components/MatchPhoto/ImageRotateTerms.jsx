// @flow
import React, { useState } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import CircleLoader from '../Loaders/CircleLoader/CircleLoader'

const baseClass = 'image-rotate-terms-modal'
const wrapperClass = `${baseClass}__wrapper`
const wrapperContainerClass = `${wrapperClass}__container`
const wrapperContainerInactiveClass = `${wrapperContainerClass}--inactive`
const wrapperToolsClass = `${wrapperClass}__tools`
const wrapperToolsMessageClass = `${wrapperToolsClass}__message`
const wrapperToolsRotateArrowClass = `${wrapperToolsClass}__rotate-arrow`
const wrapperAgreeTermsClass = `${wrapperClass}__agree-terms`
const wrapperAgreeTermsCheckboxLabelClass = `${wrapperAgreeTermsClass}__checkbox-label`
const wrapperAgreeTermsTextClass = `${wrapperAgreeTermsClass}__text`
const wrapperAgreeTermsAcceptClass = `${wrapperAgreeTermsClass}__accept`
const wrapperAgreeTermsAcceptActiveClass = `${wrapperAgreeTermsAcceptClass}--active`
const wrapperLoaderClass = `${wrapperClass}__loader`

type Props = {
  rotateImage: Function,
  createColorPins: Function,
  imageData: Object
}

const KEY_CODE_ENTER = 13
const KEY_CODE_SPACE = 32

const ImageRotateTerms = ({ rotateImage, createColorPins, imageData }: Props) => {
  const [accetTerms, setAcceptTerms] = useState(false)
  const [hideModal, setHideModal] = useState(false)
  let mouseDown = false

  function handleChange () {
    setAcceptTerms(!accetTerms)
  }

  function mouseDownHandler () {
    mouseDown = true
  }

  function mouseUpHandler () {
    setTimeout(() => {
      mouseDown = false
    }, 200)
  }

  function focusHandler (e: Object) {
    if (mouseDown) {
      e.target.blur()
    }
  }

  function keyDownHandler (e: KeyboardEvent) {
    if (e.keyCode && (e.keyCode === KEY_CODE_ENTER || e.keyCode === KEY_CODE_SPACE)) {
      e.preventDefault()
      handleChange()
    }
  }

  function clickHandler () {
    setHideModal(true)
    createColorPins(imageData)
  }

  return (
    <div className={`${wrapperClass}`}>
      <div className={`${wrapperContainerClass} ${hideModal ? `${wrapperContainerInactiveClass}` : ``}`}>
        <div className={`${wrapperToolsClass}`}>
          <div className={`${wrapperToolsMessageClass}`}>
            Use these arrows to rotate your image.
          </div>
          <button
            onMouseDown={mouseDownHandler}
            onMouseUp={mouseUpHandler}
            onFocus={focusHandler}
            className={`${wrapperToolsRotateArrowClass}`} onClick={() => rotateImage(false)}><FontAwesomeIcon icon={['fal', 'undo']} size='xs' /></button>
          <button
            onMouseDown={mouseDownHandler}
            onMouseUp={mouseUpHandler}
            onFocus={focusHandler}
            className={`${wrapperToolsRotateArrowClass}`} onClick={() => rotateImage(true)}><FontAwesomeIcon icon={['fal', 'redo']} size='xs' /></button>
        </div>
        <div className={`${wrapperAgreeTermsClass}`}>
          <div>
            <span>
              <label
                onMouseDown={mouseDownHandler}
                onMouseUp={mouseUpHandler}
                onFocus={focusHandler}
                onKeyDown={keyDownHandler}
                tabIndex='0' className={`${wrapperAgreeTermsCheckboxLabelClass}`}>
                {
                  (accetTerms) ? <FontAwesomeIcon icon={['fa', 'dot-circle']} style={{ color: '#2cabe1' }} size='lg' />
                    : <FontAwesomeIcon icon={['fa', 'dot-circle']} style={{ color: '#e5e5e5' }} size='lg' />
                }
                <input tabIndex='-1' className='visually-hidden' type='checkbox' value='terms' checked={accetTerms} onChange={handleChange} />
              </label>
            </span> <span className={`${wrapperAgreeTermsTextClass}`}>I accept Terms of Use</span>
          </div>
        </div>
        <button
          className={`${wrapperAgreeTermsAcceptClass} ${accetTerms ? `${wrapperAgreeTermsAcceptActiveClass}` : ``}`}
          onClick={() => (accetTerms) ? clickHandler() : {}}>DONE</button>
      </div>
      { hideModal && <div className={`${wrapperLoaderClass}`}><CircleLoader /></div> }
    </div>
  )
}

export {
  wrapperToolsMessageClass,
  wrapperToolsRotateArrowClass,
  wrapperAgreeTermsAcceptClass,
  wrapperAgreeTermsCheckboxLabelClass,
  wrapperAgreeTermsTextClass,
  wrapperAgreeTermsAcceptActiveClass
}
export default ImageRotateTerms
