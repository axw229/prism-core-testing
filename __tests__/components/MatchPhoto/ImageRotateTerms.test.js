/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import ImageRotateTerms, {
  wrapperToolsMessageClass,
  wrapperToolsRotateArrowClass,
  wrapperAgreeTermsAcceptClass,
  wrapperAgreeTermsCheckboxLabelClass,
  wrapperAgreeTermsTextClass,
  wrapperAgreeTermsAcceptActiveClass
} from 'src/components/MatchPhoto/ImageRotateTerms'
import CircleLoader from 'src/components/Loaders/CircleLoader/CircleLoader'

const rotateImageMockFn = jest.fn()
const createColorPinsMockFn = jest.fn()
const imageData = {}

const getImageRotateTerms = props => {
  let defaultProps = {
    rotateImage: rotateImageMockFn,
    createColorPins: createColorPinsMockFn,
    imageData: imageData
  }

  let newProps = Object.assign({}, defaultProps, props)
  return shallow(<ImageRotateTerms {...newProps} />)
}

describe('ImageRotateTerms with props', () => {
  let imageRotateTerms
  beforeEach(() => {
    if (!imageRotateTerms) {
      imageRotateTerms = getImageRotateTerms()
    }
  })

  it(`should render div.${wrapperToolsMessageClass} with headerContnet`, () => {
    const headerContnet = 'Use these arrows to rotate your image.'
    expect(imageRotateTerms.find(`div.${wrapperToolsMessageClass}`).contains(headerContnet)).toBe(true)
  })

  it(`should render two button.${wrapperToolsRotateArrowClass} buttons`, () => {
    expect(imageRotateTerms.find(`button.${wrapperToolsRotateArrowClass}`)).toHaveLength(2)
  })

  it(`should render button.${wrapperAgreeTermsAcceptClass}`, () => {
    expect(imageRotateTerms.find(`button.${wrapperAgreeTermsAcceptClass}`).exists()).toBe(true)
  })

  it(`should render button.${wrapperAgreeTermsAcceptClass} with buttonContnet`, () => {
    const buttonContnet = 'DONE'
    expect(imageRotateTerms.find(`button.${wrapperAgreeTermsAcceptClass}`).contains(buttonContnet)).toBe(true)
  })

  it(`should render label.${wrapperAgreeTermsCheckboxLabelClass}`, () => {
    expect(imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass}`).exists()).toBe(true)
  })

  it(`should render label.${wrapperAgreeTermsCheckboxLabelClass} with input checkbox`, () => {
    expect(imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass} > input`).exists()).toBe(true)
  })

  it(`should render span.${wrapperAgreeTermsTextClass}`, () => {
    expect(imageRotateTerms.find(`span.${wrapperAgreeTermsTextClass}`).exists()).toBe(true)
  })

  it(`should render span.${wrapperAgreeTermsTextClass} with spanContent`, () => {
    const spanContent = 'I accept Terms of Use'
    expect(imageRotateTerms.find(`span.${wrapperAgreeTermsTextClass}`).contains(spanContent)).toBe(true)
  })

  it(`should add class ${wrapperAgreeTermsAcceptActiveClass} to button.${wrapperAgreeTermsAcceptClass} when checkbox is checked`, () => {
    imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass} > input`).simulate('change')
    expect(imageRotateTerms.find(`button.${wrapperAgreeTermsAcceptActiveClass}`).exists()).toBe(true)
    imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass} > input`).simulate('change')
  })

  it(`should call rotateImageMockFn function when button.${wrapperToolsRotateArrowClass} is clicked`, () => {
    imageRotateTerms.find(`button.${wrapperToolsRotateArrowClass}`).map(el => {
      el.simulate('click')
      expect(rotateImageMockFn).toHaveBeenCalled()
    })
  })

  it(`should call createColorPinsMockFn function when button.${wrapperAgreeTermsAcceptClass} is clicked while checkbox is checked`, () => {
    imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass} > input`).simulate('change')
    imageRotateTerms.find(`button.${wrapperAgreeTermsAcceptClass}`).simulate('click')
    expect(createColorPinsMockFn).toHaveBeenCalled()
    imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass} > input`).simulate('change')
  })

  it(`should call createColorPinsMockFn function with prop imageData when button.${wrapperAgreeTermsAcceptClass} is clicked while checkbox is checked`, () => {
    imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass} > input`).simulate('change')
    imageRotateTerms.find(`button.${wrapperAgreeTermsAcceptClass}`).simulate('click')
    expect(createColorPinsMockFn).toHaveBeenCalledWith(imageData)
    imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass} > input`).simulate('change')
  })

  it(`should render CircleLoader when button.${wrapperAgreeTermsAcceptClass} is clicked while checkbox is checked`, () => {
    imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass} > input`).simulate('change')
    imageRotateTerms.find(`button.${wrapperAgreeTermsAcceptClass}`).simulate('click')
    expect(imageRotateTerms.find(CircleLoader).exists()).toBe(true)
    imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass} > input`).simulate('change')
  })

  it(`should add class ${wrapperAgreeTermsAcceptActiveClass} to button.${wrapperAgreeTermsAcceptClass} when checkbox is checked while space key is pressed on label`, () => {
    imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass}`).simulate('keyDown', { keyCode: 32, preventDefault: jest.fn() })
    expect(imageRotateTerms.find(`button.${wrapperAgreeTermsAcceptActiveClass}`).exists()).toBe(true)
    imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass} > input`).simulate('change')
  })

  it(`should add class ${wrapperAgreeTermsAcceptActiveClass} to button.${wrapperAgreeTermsAcceptClass} when checkbox is checked while enter key is pressed on label`, () => {
    imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass}`).simulate('keyDown', { keyCode: 13, preventDefault: jest.fn() })
    expect(imageRotateTerms.find(`button.${wrapperAgreeTermsAcceptActiveClass}`).exists()).toBe(true)
    imageRotateTerms.find(`label.${wrapperAgreeTermsCheckboxLabelClass} > input`).simulate('change')
  })
})
