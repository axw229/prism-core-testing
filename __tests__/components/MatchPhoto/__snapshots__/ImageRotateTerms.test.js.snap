// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageRotateTerms with props should match snapshot with props 1`] = `
<div
  className="image-rotate-terms-modal__wrapper"
>
  <div
    className="image-rotate-terms-modal__wrapper__container "
  >
    <div
      className="image-rotate-terms-modal__wrapper__tools"
    >
      <div
        className="image-rotate-terms-modal__wrapper__tools__message"
      >
        Use these arrows to rotate your image.
      </div>
      <button
        className="image-rotate-terms-modal__wrapper__tools__rotate-arrow"
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fal",
              "undo",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="xs"
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
      <button
        className="image-rotate-terms-modal__wrapper__tools__rotate-arrow"
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fal",
              "redo",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="xs"
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
    <div
      className="image-rotate-terms-modal__wrapper__agree-terms"
    >
      <div>
        <span>
          <label
            className="image-rotate-terms-modal__wrapper__agree-terms__checkbox-label"
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            tabIndex="0"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Array [
                  "fa",
                  "dot-circle",
                ]
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size="lg"
              spin={false}
              style={
                Object {
                  "color": "#e5e5e5",
                }
              }
              symbol={false}
              title=""
              transform={null}
            />
            <input
              checked={false}
              className="visually-hidden"
              onChange={[Function]}
              tabIndex="-1"
              type="checkbox"
              value="terms"
            />
          </label>
        </span>
         
        <span
          className="image-rotate-terms-modal__wrapper__agree-terms__text"
        >
          I accept Terms of Use
        </span>
      </div>
    </div>
    <button
      className="image-rotate-terms-modal__wrapper__agree-terms__accept "
      onClick={[Function]}
    >
      DONE
    </button>
  </div>
</div>
`;
